File: Integration Examples/eAPI v1.6/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/PayOneclickInitReq.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects
{
    public class PayOneclickInitReq : SignBase, IBaseRequest
    {
        [JsonProperty("merchantId")]
        public string MerchantId;

        [JsonProperty("origPayId")]
        public string OrigPayId { get; set; }

        [JsonProperty("orderNo")]
        public string OrderNo { get; set; }

        [JsonProperty("totalAmount")]
        public long TotalAmount { get; set; }

        [JsonProperty("currency")]
        public string Currency { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }

        public string ToSign()
        {
            var sb = new StringBuilder();
            Add(sb, MerchantId);
            Add(sb, OrigPayId);
            Add(sb, OrderNo);
            Add(sb, DateTime);
            Add(sb, TotalAmount);
            Add(sb, Currency);
            Add(sb, Description);
            DeleteLast(sb);
            return sb.ToString();
        }
    }
}

File: Integration Examples/eAPI v1.7/java/eAPI-v17-connector/src/main/java/cz/monetplus/mips/eapi/v17/connector/entity/PayOneclickInitReq.java
package cz.monetplus.mips.eapi.v17.connector.entity;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class PayOneclickInitReq extends SignBase {

	private static final long serialVersionUID = -438284924457338635L;

	@XmlElement(nillable=false)
	public String merchantId;
	
	@XmlElement(nillable=false)
	public String origPayId;

	@XmlElement(nillable=false)
	public String orderNo;

	@XmlElement(nillable=true)
	public Long totalAmount;

	@XmlElement(nillable=true)
	public String currency;

	@XmlElement(nillable=true)
	public String description;

	@Override
	public String toSign() {
		StringBuilder sb = new StringBuilder(); 
		add(sb, merchantId);
		add(sb, origPayId);
		add(sb, orderNo);
		add(sb, dttm);
		add(sb, totalAmount);
		add(sb, currency);
		add(sb, description);
		deleteLast(sb);
		return sb.toString();
	}

}
